Project benefits content (#1085)
dotnet-foundation/website: input/projects/why-join-wip.md

# Why should my project join the .NET Foundation?

> The .NET Foundation offers administrative and technical services to open source projects, to help maintainers govern and develop their projects.

Reviewer: sentence splice
```suggestion
The .NET Foundation offers administrative and technical services to open source projects to help maintainers govern and develop their projects.
```

.NET Foundation services aim to increase project stability and maturity and don't replace developers on member projects. Day-to-day development remains in the hands of project maintainers and contributors.

Project benefits include:

## Project guidance and mentoring

> New projects joining us will get mentorship on how to best run an open source community along with access to developers currently seeking .NET projects. We work with project leads to help ensure the project grows into a vibrant and welcoming community. And, of course, we love to share best practices for managing a friendly, collaborative, and innovative open source workplace, wherever you may be.

Reviewer 2: **QUESTION:** Could you give examples of mentoring and a vibrant community?

Reviewer 3: The mentoring has come during the maintainer's committee meetings as something they'd like to adopt in the near future.

Author: I can't give examples, I'm just playing around with content.

I think some more detail from people who know more about this area would be useful.

Reviewer 3: @JamesNK it's being discussed as something we'd like to start adopting in the maintainer's committee but not something currently that's provided beyond the initial onboarding help with the infrastructure.

## IP and legal

> Legal can be tricky. We can help. Projects in our organization will get assistance to organize and manage their IP assignments as well as infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects. We can also help ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation. Need advice on legal issues? We offer support there as well.

Author: I don't think examples are a good fit for a summary page like this.

> Legal can be tricky. We can help. Projects in our organization will get assistance to organize and manage their IP assignments as well as infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects. We can also help ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation. Need advice on legal issues? We offer support there as well.

Reviewer 2: I didn't suggest the particular project example, rather giving the range of services. I'm in the .NET Foundation for 2 years and don't know much about that, so my guess is that this will be also enigmatic for new people.

> Legal can be tricky. We can help. Projects in our organization will get assistance to organize and manage their IP assignments as well as infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects. We can also help ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation. Need advice on legal issues? We offer support there as well.

Author: The paragraph already gives examples. Would it help you if they were in a bullet point list? e.g.

> Legal can be tricky. We can help. Projects in our organization will get:
> * Assistance to organize and manage their IP assignments
> * Infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects.
> * Help to ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation.
> * Support and advice on legal issues

> Legal can be tricky. We can help. Projects in our organization will get assistance to organize and manage their IP assignments as well as infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects. We can also help ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation. Need advice on legal issues? We offer support there as well.

Author: ```suggestion
Legal can be tricky. We can help. Projects in our organization will get:

* Assistance to organize and manage their IP assignments
* Infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects.
* Help to ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation.
* Support and advice on legal issues
```

> ## Marketing and communications

Reviewer 4: This bullet point isn't exclusive to .NET Foundation membership. The .NET Foundation will promote any project related to .NET at this point.

Author: Good comment. I reached out to people and got clarification about marketing services and opportunities unique to members. Section updated.

> The more people who know about .NET, the better for the ecosystem. We work with Amazon, JetBrains, Microsoft, Progress, and other industry leaders to improve your project's exposure within the community at large. Your success is our success-we'd love to see you achieve the accolades you deserve!

Reviewer: ```suggestion
The more people who know about .NET, the better for the ecosystem. We work with Amazon, JetBrains, Microsoft, Progress, and other industry leaders to improve your project's exposure within the community at large. Your success is our success - we'd love to see you achieve the accolades you deserve!
```

> ## CLA management

Reviewer 4: The CLA is just a bot you add. .NET Foundation doesn't provide anything (except a ready made document you must use).

Reviewer: Arguably having the legalese covered is a benefit, and the thing I like most about the .NET Foundation CLA is that it applies to all DNF projects so you sign it once and it's gone away forever, which I think is alluded to in this section.

Reviewer 3: Yeah doesn't hurt to be concise I guess

A Contributor License Agreement (CLA) ensures a project has the necessary rights to code contributions.

Our automated CLA system has been working well for GitHub-based projects, with a very high rate of developers able to easily sign their Contribution License Agreements electronically. This service is offered with full automated notification to the project committers regarding the CLA status of a pull request author.

## Secret management

> Get a secure vault to store secrets related to your project (social media credentials, website logins, etc.). Share credentials securely between trusted maintainers on your project, and enjoy the peace of mind that they're safely saved

Reviewer 2: **QUESTION:** Could you expand on what technology is and how it's different from GH repository/organisation secrets?

Author: I'm guessing it is Azure Key Vault? IDK. More detail here would be good.

## Code signing

Code signing protects against content tampering. A signature guarantees the assets haven't been modified since the asset was signed.

We've implemented an automated service to Authenticode signing your release MSIs, EXEs, and NuGet packages with the .NET Foundation code signing certificate. The certificate is well known to Windows Smart Screen so that people can install your open-source application without warnings. It also allows for consuming applications knowing that the binary downloaded is the one created on your build server.

## Hosting

> We provide hosting for a number of .NET Foundation project resources in Microsoft Azure. We have a .NET Foundation Office 365 account with SharePoint Online, Exchange, and more.

Reviewer 2: **QUESTION:** How to get this benefit?

Author: I don't know. However, that sort of documentation belongs somewhere else. If you think there aren't good docs on using benefits then you could create a new issue?

## Tools

> If your maintainers need access to certain software to enable them to work more effectively on their open source projects, contact us and we can talk with vendors on your behalf. Microsoft has sponsored a number of MSDN subscriptions for .NET Foundation project contributors over the past 12 months.

Reviewer 4: @glennawatson is right. The MSDN subscription is near impossible to come by any longer, even for renewals. I would not highlight this as a benefit.

## Continuous integration

> The .NET Foundation has a dedicated Azure DevOps instance, which allows projects to use as many agents in Pipelines as they need and provide storage for CI feeds with Artifacts.

Reviewer 2: **QUESTION:** How is it different from the regular Azure DevOps free tier for OSS projects?

Author: I guess the dedicated Azure DevOps instance has higher limits on CI minutes and Artifacts storage. Getting a little more detail here would be good.

Reviewer 3: The Azure DevOps has higher limits on CI minutes/artifacts, but it's shared between multiple projects so when other projects have high usage there can be delays to projects running.

# Why should my project join the .NET Foundation?

The .NET Foundation offers administrative and technical services to open source projects to help maintainers govern and develop their projects.

> .NET Foundation services aim to increase project stability and maturity and don't replace developers on member projects. Day-to-day development remains in the hands of project maintainers and contributors.

Reviewer 5: .NET Foundation services aim to increase project stability and maturity. It does not replace developers on member projects. Day-to-day development remains in the hands of project maintainers and contributors.

Author: Thanks

> .NET Foundation services aim to increase project stability and maturity and don't replace developers on member projects. Day-to-day development remains in the hands of project maintainers and contributors.

Author: ```suggestion
.NET Foundation services aim to increase project stability and maturity. It doesn't replace developers on member projects. Day-to-day development remains in the hands of project maintainers and contributors.
```

Project benefits include:

## Project guidance and mentoring

New projects joining us will get mentorship on how to best run an open source community along with access to developers currently seeking .NET projects. We work with project leads to help ensure the project grows into a vibrant and welcoming community. And, of course, we love to share best practices for managing a friendly, collaborative, and innovative open source workplace, wherever you may be.

## IP and legal

Legal can be tricky. We can help. Projects in our organization will get:

* Assistance to organize and manage their IP assignments
* Infrastructure for tracking common, simple, Contribution License Agreements across all .NET Foundation projects.
* Help to ensure that your collaborators respect the open source licenses and trademarks of projects assigned to the .NET Foundation.
* Support and advice on legal issues

## Marketing and communications

> The more people who know about .NET, the better for the ecosystem. We work with Amazon, JetBrains, Microsoft, Progress, and other industry leaders to improve your project's exposure within the community at large. Your success is our success - we'd love to see you achieve the accolades you deserve!

Author: ```suggestion
The more people who know about .NET, the better for the ecosystem. We work with Amazon, JetBrains, Microsoft, Progress, and other industry leaders to improve your project's exposure within the community at large. Your success is our success - we'd love to see you achieve the accolades you deserve!

The .NET Foundation helps any .NET project that reaches out to it with marketing advice or promotion via social media. Additionally, the .NET Foundation helps produce videos about member projects and maintainers:

* [.NET Foundation Project Spotlight](https://www.youtube.com/playlist?list=PL1rZQsJPBU2Ql5ZBOOdqn32xPMEUdor5Z)
* [Maintainer Series](https://www.youtube.com/playlist?list=PL1rZQsJPBU2TA8lm-2S0TMdT5FlL1sxuF)
```
